ubcpi/answer_pool.py

```diff
@@ -0,0 +1,315 @@
+import random
+import persistence as sas_api
+
+POOL_SIZE = 50
+
+
+class UnknownChooseAnswerAlgorithm(Exception):
+    pass
+
+
+def offer_answer(pool, answer, rationale, student_id, algo):
+    """
+    submit a student answer to the answer pool
+
+    The answer maybe selected to stay in the pool depending on the selection algorithm
+
+    Args:
+        pool (dict): answer pool
+            Answer pool format:
+            {
+                option1_index: {
+                    'student_id': { can store algorithm specific info here },
+                    ...
+                }
+                option2_index: ...
+            }
```

Comment:
@xcompass correct me if I'm wrong here. The "answer pool" is a dictionary in the above format? What does it contain?

@nedbat and I are trying to figure out how you're representing the answer pool, how big it can get, and how it is stored.